packages/in_app_purchase/lib/src/in_app_purchase_connection/app_store_connection.dart

```diff
@@ -26,4 +27,11 @@ class AppStoreConnection implements InAppPurchaseConnection {
     _instance = AppStoreConnection();
     return _instance;
   }
+
+  @override
+  Future<List<Product>> getProductList(List<String> identifiers) =>
+      throw UnimplementedError();
+
+  Future<List<SKProductWrapper>> getSKProductList(List<String> identifiers) =>
```

Comment:
@mklim was the intention for `AppStoreConnection` and `GooglePlayConnection` to not extend the API defined in `InAppPurchaseConnection` and expose platform specific features only in the wrappers?

(I guess I'm ok with both options, just wanted to make sure I'm on the same page with you here)

Reply:
Yeah, this layer should be the generic API only. @cyanglaz, I think you can revert this file entirely. This patch can just be isolated to the `wrapper/` layer instead of building out the generic code.